Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs

```diff
@@ -647,6 +648,24 @@ unlinkDL merchantShortId driverId = do
   logTagInfo "dashboard -> unlinkDL : " (show personId)
   pure Success
 
+---------------------------------------------------------------------
+unlinkAadhaar :: ShortId DM.Merchant -> Id Common.Driver -> Flow APISuccess
+unlinkAadhaar merchantShortId driverId = do
+  merchant <- findMerchantByShortId merchantShortId
+
+  let driverId_ = cast @Common.Driver @DP.Driver driverId
+  let personId = cast @Common.Driver @DP.Person driverId
+
+  driver <- Esq.runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
```

Comment:
Just check if Aadhaar entry is there or not. If not, throw error can't unlink. Else unlink and also  make enabled=false for the driver if merchant.aadhaarRequired is true.